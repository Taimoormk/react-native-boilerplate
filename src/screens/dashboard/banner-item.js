import React from 'react';
import { string } from 'prop-types';
import {
  LabelHeading,
  LabelText,
  ListItemBanner,
  ListItemWrapper,
  TextWrapper,
} from './styled';

export const BannerItem = ({
  banner,
  description,
  label,
}) => (
  <ListItemWrapper>
    <ListItemBanner
      source={{ uri: banner }}
    />
    <TextWrapper>
      <LabelHeading>{label}</LabelHeading>
      <LabelText>{description}</LabelText>
    </TextWrapper>
  </ListItemWrapper>
);

BannerItem.propTypes = {
  banner: string.isRequired,
  description: string.isRequired,
  label: string.isRequired,
};
BannerItem.defaultProps = {};
