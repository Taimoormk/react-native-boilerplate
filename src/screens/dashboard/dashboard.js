import {
  TextInput,
} from '@taimoormk/react-native-commons-collection/text-input';
import {
  Image,
} from '@taimoormk/react-native-commons-collection/image';
import React from 'react';
import { View, Text } from 'react-native';
import { Button } from 'react-native-paper';
import { styles } from './styles';
import { STYLE_CONSTANTS } from '../../globals/styles';

export const Dashboard = () => (
  <View style={styles.container}>
    <Text style={styles.heading}>Dashboard</Text>
    <TextInput />
    <Button
      mode="contained"
      color={STYLE_CONSTANTS.COLORS.FACEBOOK}
      style={styles.facebookButton}
    >
        Hello
      </Button>
    <Image source={{ uri: 'https://facebook.github.io/react-native/img/tiny_logo.png' }} />
    <Text style={styles.subHeading}>Hello</Text>
  </View>
);
