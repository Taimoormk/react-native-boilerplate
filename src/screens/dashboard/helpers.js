export const openSourceProjects = [
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1565548545/GitHub/React_Pakistan_Storybook_Logo.jpg',
    id: 'react-pakistan-open-source-project-1',
    label: 'React Pakistan',
  },
  {
    banner: 'https://camo.githubusercontent.com/b3338a9d26ce4dbe5b5e4025c27f4761adafa058/68747470733a2f2f692e6962622e636f2f6e6e32583177522f524e426f696c6572706c617465732e6a7067',
    id: 'react-pakistan-open-source-project-2',
    label: 'React Native Boilerplates',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650645/GitHub/react-native-commons-collection.jpg',
    id: 'react-pakistan-open-source-project-3',
    label: 'React Native Commons Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650649/GitHub/react-native-ui-collection.jpg',
    id: 'react-pakistan-open-source-project-4',
    label: 'React Native UI Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    id: 'react-pakistan-open-source-project-5',
    label: 'React Native Icon Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1583650647/GitHub/react-native-icon-collection.jpg',
    id: 'react-pakistan-open-source-project-6',
    label: 'React Native Logo Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1566355017/GitHub/react-commons-collection.jpg',
    id: 'react-pakistan-open-source-project-7',
    label: 'React Commons Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1566355010/GitHub/react-ui-collection.jpg',
    id: 'react-pakistan-open-source-project-8',
    label: 'React UI Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499955/GitHub/react-icons-collection.jpg',
    id: 'react-pakistan-open-source-project-9',
    label: 'React Icon Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499957/GitHub/react-logo-collection.jpg',
    id: 'react-pakistan-open-source-project-10',
    label: 'React Logo Collection',
  },
  {
    banner: 'https://res.cloudinary.com/dq6hflqwx/image/upload/v1567499950/GitHub/react-design-collection.jpg',
    id: 'react-pakistan-open-source-project-11',
    label: 'React Design Collection',
  },
];
