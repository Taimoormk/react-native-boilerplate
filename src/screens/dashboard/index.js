export { Dashboard } from './dashboard';
export { DashboardMobile } from './components/mobile';
export { DashboardWeb } from './components/web';
