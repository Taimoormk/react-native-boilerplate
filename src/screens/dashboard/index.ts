export * from './dashboard';
export * from './mobile';
export * from './web';
