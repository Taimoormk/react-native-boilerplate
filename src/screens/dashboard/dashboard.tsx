import { IRNTheme } from '@react-pakistan/util-react-native-functions';
import React, { ReactElement, memo } from 'react';
import { FlatList } from 'react-native';
import { withTheme } from 'styled-components/native';
import { BannerItem } from '../../components/banner-item';
import { dashboardScreenText, openSourceMainProjects } from './helpers';
import {
  DashboardHeading,
  DashboardWrapper,
  ListHeadingWrapper,
} from './styled';

export const Dashboard = memo(withTheme(({
  theme,
} : IDashboardProps) : ReactElement => {
  const renderItem = ({ item: { banner, description, label } } : any) : ReactElement => (
    <BannerItem
      banner={banner}
      description={description}
      label={label}
    />
  );

  return (
    <DashboardWrapper>
      <FlatList
        data={openSourceMainProjects}
        keyExtractor={({ id } : any) : string => id}
        ListHeaderComponent={
          <ListHeadingWrapper>
            <DashboardHeading
              {...theme.typography.h1}
            >
              {dashboardScreenText.mainHeading}
            </DashboardHeading>
          </ListHeadingWrapper>
        }
        renderItem={renderItem}
      />
    </DashboardWrapper>
  );
}));

export interface IDashboardProps {
  /**
   *
   */
  theme : IRNTheme;
}
