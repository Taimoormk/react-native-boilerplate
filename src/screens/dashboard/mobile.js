// ########## Import Dependencies Here ##########
import { FlatList } from '@react-pakistan/react-native-commons-collection/flat-list';
import { object } from 'prop-types';
import React, { memo } from 'react';
import { withTheme } from 'styled-components';

// ########## Import Components Here ##########
import { BannerItem } from '../../components/banner-item';
import { appTheme } from '../../theme';
import { dashboardScreenText, openSourceMobileProjects } from './helpers';
import {
  DashboardHeading,
  DashboardWrapper,
  ListHeadingWrapper,
} from './styled';

export const DashboardMobile = memo(withTheme(({
  theme,
}) => {
  const renderItem = ({ item: { banner, description, label } }) => (
    <BannerItem
      banner={banner}
      description={description}
      label={label}
    />
  );

  return (
    <DashboardWrapper>
      <FlatList
        data={openSourceMobileProjects}
        keyExtractor={({ id }) => id}
        ListHeaderComponent={
          <ListHeadingWrapper>
            <DashboardHeading
              {...theme.typography.h1}
            >
              {dashboardScreenText.mainHeading}
            </DashboardHeading>
          </ListHeadingWrapper>
        }
        renderItem={renderItem}
      />
    </DashboardWrapper>
  );
}));

DashboardMobile.propType = {
  theme: object.isRequired,
};
DashboardMobile.defaultProps = {
  theme: appTheme,
};
