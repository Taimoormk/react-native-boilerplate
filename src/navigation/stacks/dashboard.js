import { createStackNavigator } from '@react-navigation/stack';
import { Icon } from '@react-pakistan/react-native-icon-collection/icon';
import { shape, func } from 'prop-types';
import React from 'react';
import { useDispatch } from 'react-redux';
import * as actions from '../../redux/actions';
import { Dashboard as DashboardScreen } from '../../screens/dashboard';
import { resolveToSentenceCase } from '../../utils';
import { NAVIGATION_ROUTES, NAVIGATION_TITLES } from '../navigation-routes';
import { HeaderIcon } from '../styled';

const Stack = createStackNavigator();

export const DashboardStack = ({
  navigation,
}) => {
  const dispatch = useDispatch();

  return (
    <Stack.Navigator>
      <Stack.Screen
        component={DashboardScreen}
        name={NAVIGATION_ROUTES.TAB_MAIN}
        options={{
          headerRight: () => (
            <HeaderIcon
              onPress={() => dispatch(actions.hideAppAction())}
            >
              <Icon
                icon='MobileUiGrey9'
              />
            </HeaderIcon>
          ),
          headerLeft: () => (
            <HeaderIcon
              onPress={() => navigation.toggleDrawer()}
            >
              <Icon
                icon='MobileUiGrey15'
              />
            </HeaderIcon>
          ),
          headerTitle: resolveToSentenceCase(NAVIGATION_TITLES.TAB_MAIN),
        }}
      />
    </Stack.Navigator>
  );
};

DashboardStack.propTypes = {
  navigation: shape({
    toggleDrawer: func.isRequired,
  }).isRequired,
};
