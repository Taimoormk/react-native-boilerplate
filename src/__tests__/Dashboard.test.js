// ########## Import Dependencies Here ##########
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

// ########## Import Components Here ##########
import { Dashboard } from '../screens/dashboard';

test('renders Dashboard component correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<Dashboard />);
  const wrapper = renderer.getRenderOutput();
  expect(wrapper).toMatchSnapshot();
});
