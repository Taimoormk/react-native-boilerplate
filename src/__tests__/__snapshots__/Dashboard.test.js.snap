// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Dashboard component correctly 1`] = `<Unknown />`;
